cfme/markers/env_markers/provider.py

```diff
@@ -334,24 +334,30 @@ def add_prov(prov):
         argvalues.append(pytest.param(data_prov))
 
         # Use the provider key for idlist, helps with readable parametrized test output
-        ver = data_prov.version if data_prov.version else None
-        if ver:
-            the_id = "{}-{}".format(data_prov.type_name, data_prov.version)
+        if metafunc.config.getoption('legacy_ids'):
+            the_id = "{}".format(data_prov.key)
         else:
-            the_id = "{}".format(data_prov.type_name)
+            ver = data_prov.version if data_prov.version else None
+            if ver:
+                the_id = "{}-{}".format(data_prov.type_name, data_prov.version)
+            else:
+                the_id = "{}".format(data_prov.type_name)
 
         # Now we modify the id based on what selector we chose
-        if selector == ONE:
-            if need_prov_keys:
+        if metafunc.config.getoption('disable_selectors'):
```

Comment:
@psav To be honest, I am not quite sure what that CLI option does or does not. Could you maybe send an e-mail explaining the usage of this once this is merged?